### PR TITLE
Reactjs week5

### PR DESCRIPTION
**Pull request by kfazinic**

### CODE REVIEW

File: react/week5/shift-app/client/src/components/LoginModal.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        className="login-btn btn"
+        onClick={async e => {
+          e.preventDefault();
+          const user = await Firebase().signInGoogle();
+          setUserState(user);
+        }}
+      >
+        Sign in with Google
+      </button>
+    </Modal>
```
Comment (senner007) [inline react/week5/shift-app/client/src/components/LoginModal.js]: Hi :)
I'm thinking the modal should close automatically after login

File: react/week5/shift-app/client/src/App.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        }
+      });
+  }, []);
+
+  useEffect(() => {
+    const fetchIfAuthorized = async () => {
+      if (!!userState) {
+        getPrivateContent()
+          .then(r => {
+            if (r.body === "Token verified") {
```
Comment (senner007) [inline react/week5/shift-app/client/src/App.js]: This won't work. 
You can still make a call to "http://localhost:5000/shifts" without a valid token
You have to verify the auth token on each request

File: react/week5/shift-app/client/src/App.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              fetch("http://localhost:5000/shifts")
```
Comment (senner007) [inline react/week5/shift-app/client/src/App.js]: Try and see if you can avoid indentation. It makes the code much harder to read. 
Always imagine that someone else has to be able to understand your code immediately. Also programmers and others who are not used to javascript

File: react/week5/shift-app/client/src/components/LoginModal.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import React, { useState, useContext } from "react";
+import UserContext from "../App";
+import { Modal, Button } from "react-materialize";
+import Firebase from "../auth/firebase";
+
+export default function LoginModal({ setUserState }) {
```
Comment (senner007) [inline react/week5/shift-app/client/src/components/LoginModal.js]: Also avoid indentation and large components.
To quote Robert C. Martin:
"Large functions are where classes go to hide"
or to put it in context :
"Large React components are where other components go to hide"

File: react/week5/shift-app/client/src/components/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import React, { useContext } from "react";
+import UserContext from "../UserContext";
+import firebaseInst from "../auth/firebase";
+import { TextInput } from "react-materialize";
+
+export default function Navbar({ setUserState, searchValue, setSearchValue }) {
+  const userState = useContext(UserContext);
```
Comment (senner007) [inline react/week5/shift-app/client/src/components/Navbar.js]: Again. Try to extract the logic into smaller components and functions. Each function and component should be completely self-explanatory to the reader

File: react/week5/shift-app/client/src/App.js (hunk 2 above)
Comment (kfazinic) [reply react/week5/shift-app/client/src/App.js]: Thanks! Fixed 😅

File: react/week5/shift-app/client/src/components/LoginModal.js (hunk 1 above)
Comment (kfazinic) [reply react/week5/shift-app/client/src/components/LoginModal.js]: Fixed :)

**Pull request merged**